back/engines/commercial/verification/app/services/verification/patches/permissions/user_requirements_service.rb

```diff
@@ -40,7 +40,13 @@ def mark_satisfied_requirements!(requirements, permission, user)
           super
 
           if user.verified?
-            requirements[:authentication][:missing_user_attributes] = [] if permission.permitted_by == 'verified'
+            if permission.permitted_by == 'verified'
+              requirements[:authentication][:missing_user_attributes] = if user.email.present? && user.confirmation_required?
```

Comment:
Nit: It's maybe not ideal to have 3 nested ifs, but I don't really see how the code could be improved so definitely fine to keep this.